src/Features/CSharp/Portable/UseCollectionInitializer/CSharpUseCollectionInitializerCodeFixProvider.cs

```diff
@@ -100,7 +100,14 @@ private static ExpressionSyntax ConvertInvocation(InvocationExpressionSyntax inv
 
             if (arguments.Count == 1)
             {
-                return arguments[0].Expression;
+                // Assignment expressions in a collection initializer will cause the compiler to 
+                // report an error.  This is because { a = b } is teh form for an object initializer,
+                // and the two forms are not allowed to mix/match.  Parenthesize the assignment to
+                // avoid the ambiguity.
+                var expression = arguments[0].Expression;
```

Comment:
multi argument is when ou have:

```c#
dict = new Dictionary<int, string>();
dict.Add(1, 2);
//
this gets converted to:

dict = new Dictionary<int, string>
{ 
    { 1, 2 }
}
```

So there's no ambiguity with assignment expressions like there might be when you have an arbitrary expression.